### PR TITLE
EA-2190 delete usersets associated to user

### PR DESCRIPTION
**Pull request by SrishtiSingh-eu**

### CODE REVIEW

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 1]
```diff
@@ -724,5 +725,58 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet updateUserSet, L
...
+	try {
+		// get the user from authentication
+		Agent user = new WebUser();
+		user.setHttpUrl(getUserSetService().getUserId(authentication));
+		user.setNickname(((ApiCredentials)authentication.getCredentials()).getUserName());
+
+		List<PersistentUserSet> userSets = getUserSetService().getUserSetByCreatorId(user.getHttpUrl());
+
+		// generate “ETag”;
+		String eTagNew = generateETag(userSets.get(0).getModified(), WebFields.FORMAT_JSONLD, getApiVersion());
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:757]: the method doesn't return an eTag

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 2]
```diff
@@ -724,5 +725,58 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet updateUserSet, L
...
+		// if the user set is disabled and the user is not an admin, respond with HTTP
+		// 410
+		HttpStatus httpStatus = null;
+		httpStatus = HttpStatus.NO_CONTENT;
+
+		// delete user sets
+		getUserSetService().deleteUserSets(userSets);
+
+		// build response entity with headers
+		MultiValueMap<String, String> headers = new LinkedMultiValueMap<>(5);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:768]: The method must return only the following headers

headers.add(HttpHeaders.ALLOW, UserSetHttpHeaders.ALLOW_PGD);
	    headers.add(UserSetHttpHeaders.CACHE_CONTROL, UserSetHttpHeaders.VALUE_NO_CAHCHE_STORE_REVALIDATE);
	    
please create, and use it in DELETE and POST "/set/" methods
UserSetHttpHeaders.ALLOW_PGD = "POST,GET,DELETE";

File: set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java [hunk 3]
```diff
@@ -83,7 +83,17 @@ public UserSet getUserSetById(String userSetId) throws UserSetNotFoundException
 	return res;
     }
 
-    /**
+	@Override
+	public List<PersistentUserSet> getUserSetByCreatorId(String creatorId) throws UserSetNotFoundException {
+		List<PersistentUserSet> res = getMongoPersistence().getByCreator(creatorId).asList();
+		if (res.isEmpty()) {
+			throw new UserSetNotFoundException(UserSetI18nConstants.USERSET_NOT_FOUND_FOR_USER,
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java:90]: no exception should be thrown if there are no sets created by an user

**Review by gsergiu — CHANGES_REQUESTED** [review]
please implement the requested changes

File: set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java [hunk 4]
```diff
@@ -252,6 +259,16 @@ public void remove(String id) {
 		    getDao().delete(userSet);
 	}
 
+	@Override
+	public void removeAll(List<PersistentUserSet> userSets) {
+		if (!userSets.isEmpty()) {
+			for( PersistentUserSet userSet : userSets) {
+				getDao().delete(userSet);
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java:266]: have created a new method in PersistentUserSetDaoImpl which deletes using objectId. I have used 
<T> WriteResult delete(Query<T> var1);  
as the above-mentioned one throws an error and asks for disabling Validation, Which is not available in that method.

File: set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/SetControllerTest.java [hunk 5]
```diff
@@ -106,4 +107,25 @@ public void create_UserSet_400_unauthorized_InvalidJWTToken() throws Exception {
...
+                .header(HttpHeaders.AUTHORIZATION, token)
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE))
+//		.andDo(print())
+                .andExpect(status().is(HttpStatus.NO_CONTENT.value()));
+    }
+
+    @Test
+    public void testDeleteUserAssociatedSets_NotAuthorised() throws Exception {
+
+        mockMvc.perform(delete(BASE_URL + "delete/")
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/SetControllerTest.java:124]: changed

File: set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/SetControllerTest.java [hunk 6]
```diff
@@ -106,4 +107,25 @@ public void create_UserSet_400_unauthorized_InvalidJWTToken() throws Exception {
 		.andExpect(status().is(HttpStatus.UNAUTHORIZED.value()));	
     }
 
+
+    @Test
+    public void testDeleteUserAssociatedSets_Success() throws Exception {
+
+        mockMvc.perform(delete(BASE_URL + "delete/")
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/SetControllerTest.java:114]: changed

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 7]
```diff
@@ -724,5 +725,58 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet updateUserSet, L
 	    throw new InternalServerException(e);
 	}
     }
+	@DeleteMapping(value = { "/set/delete/" })
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:728]: changed

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 8]
```diff
@@ -724,5 +725,58 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet updateUserSet, L
...
+	/**
+	 * This method implements removal of a all sets associated to a user
+	 *
+	 * @param authentication
+	 * @throws HttpException
+	 */
+	protected ResponseEntity<String> deleteUserAssociatedSets(Authentication authentication) throws HttpException {
+	try {
+		// get the user from authentication
+		Agent user = new WebUser();
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:750]: changed

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 9]
```diff
@@ -724,5 +725,58 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet updateUserSet, L
...
+		// delete user sets
+		getUserSetService().deleteUserSets(userSets);
+
+		// build response entity with headers
+		MultiValueMap<String, String> headers = new LinkedMultiValueMap<>(5);
+		headers.add(HttpHeaders.LINK, UserSetHttpHeaders.VALUE_BASIC_CONTAINER);
+		headers.add(HttpHeaders.LINK, UserSetHttpHeaders.VALUE_BASIC_RESOURCE);
+		headers.add(HttpHeaders.ALLOW, UserSetHttpHeaders.ALLOW_GPD);
+		headers.add(UserSetHttpHeaders.VARY, HttpHeaders.PREFER);
+		headers.add(UserSetHttpHeaders.ETAG, eTagNew);
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:773]: removed etag

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 1 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:757]: removed etag

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 2 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:768]: have added ALLOW_PGD = "POST,GET,DELETE"; in UserSetHttpHeaders. have modified the headers

File: set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java (hunk 3 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java:90]: I will remove the exception

File: set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java:266]: Have created a method deleteByObjectId(List<ObjectId> objectIds) in PersistentUserSetDaoImpl to implement delete(final Class clazz, final Iterable ids)

**Pull request merged**